src/binaryen-c.cpp

```diff
@@ -273,6 +273,45 @@ BinaryenType BinaryenTypeExnref(void) { return exnref; }
 BinaryenType BinaryenTypeUnreachable(void) { return unreachable; }
 BinaryenType BinaryenTypeAuto(void) { return uint32_t(-1); }
 
+BinaryenType BinaryenTypeCreate(BinaryenType* valueTypes, uint32_t numTypes) {
+  std::vector<Type::ValueType> types;
+  types.reserve(numTypes);
+  for (size_t i = 0; i < numTypes; ++i) {
+    assert(Type(valueTypes[i]).isValueType());
+    types.push_back(static_cast<Type::ValueType>(valueTypes[i]));
+  }
+  Type result(types);
+
+  if (tracing) {
+    std::string array = getTemp();
+    std::cout << "  {\n";
+    std::cout << "    BinaryenType " << array << "[] = {";
+    for (size_t i = 0; i < numTypes; ++i) {
+      std::cout << int(valueTypes[i]);
+      if (i < numTypes - 1) {
+        std::cout << ", ";
+      }
+    }
+    std::cout << "};\n";
+    std::cout << "    BinaryenTypeCreate(" << array << ", " << numTypes
+              << "); // " << int(result) << "\n";
+    std::cout << "  }\n";
+  }
+
+  return uint32_t(int(result));
```

Comment:
Why double cast? Isn't `Type`s cast operand to `uint32_t` in the first place?

Reply:
Thanks for catching these. I originally had the implicit conversion go to int and didn't update these when I changed it.